### DISCUSSION TITLE
New edge function has undefined global connection

### DISCUSSION THREAD

**Discussion by Clancey** [sqlitecloud/discussions#16] [Bug Report] [answered]
I created a new account, setup a database. And added an Edge Function. When I run it in the dashboard using the test button, everything works great. However when I go to the url I get an error:
```
{
    "error": {
        "status": "500",
        "title": "TypeError: undefined is not an object (evaluating 'connection')",
        "detail": "GET /v2/functions/RandomPuzzle an error occoured"
    }
}
```

**Comment by marcobambini** [comment]
Hi, @Clancey, have you assigned an APIKEY to your function?
In test mode, we use the admin APIKEY to authenticate the connection, but in production, you need to have an APIKEY assigned (or passed as a parameter) to your function for security reasons.

**Reply by Clancey** [reply, marked as answer]
That was it! Maybe the error message should mention the lack of an API key to nudge people in the right direction?

**Comment by marcobambini** [comment]
You are right. We are working on the documentation, which should be available within a few days, and we'll improve the returned error message.
